.changeset/khaki-scissors-look.md

```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/graphql-client': patch
```

Comment:
Since we're adding support for `signal`, would this make more sense to be a `minor` update?

Reply:
I originally had it as minor actually, but decided against it upon further thought. Since this is a type-only change, this PR is actually a complete no-op in terms of the code that we ship to NPM.

There is a fair argument to be made that this should be minor because we are changing the public API, although technically the same outcome with passing through `signal` could have already been done by JS consumers of this library anyway.

Reply:
Given we have since debunked that there won't be a non-types change to ship, this is now being changed to a minor version bump.